misk-hibernate/src/main/kotlin/misk/vitess/StartVitessService.kt

```diff
@@ -347,9 +343,41 @@ class StartVitessService(
 
   fun cluster() = cluster!!.cluster
 
+  fun shouldRunVitess() =  config.type != DataSourceType.VITESS || environment == TESTING || environment == DEVELOPMENT
+
   override fun shutDown() {
   }
 
+  /**
+   * Shut down the cached clusters on JVM exit.
+   */
+  init {
+    // We need to do this outside of the service start up because this takes a really long time
+    // the first time you do it and can cause service manager to time out.
+    if (shouldRunVitess() && imagePulled.compareAndSet(false, true) &&
+        runCommand("docker pull vitess/base@$VITESS_VERSION") != 0) {
+      logger.warn("Failed to pull Vitess docker image. Proceeding regardless.")
+    }
+    Runtime.getRuntime().addShutdownHook(Thread {
```

Comment:
This now just gets run a ton of times? I guess it'll just be a noop

Reply:
hmm good point. I can move that back to the companion init